File: leetcode/0075-sort-colors/solution.java
class Solution {
    public void sortColors(int[] nums) {
        int zeroPointer = 0, twoPointer = nums.length - 1, index = 0;
        while(index <= twoPointer){
            if(nums[index] == 0){
                swap(nums, zeroPointer, index);
                zeroPointer++;
                index++;
            }
            else if(nums[index] == 2){
                swap(nums, twoPointer, index);
                twoPointer--;
            }
            else{
                index++;
            }
        }
    }
    public void swap(int[] nums, int n1, int n2){
        int temp = nums[n1];
        nums[n1] = nums[n2];
        nums[n2] = temp;
    }
}

File: leetcode/0075-sort-colors/solution.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        l, r = 0, len(nums) - 1
        i = 0

        while i <= r:
            if nums[i] == 2 and i < r:
                nums[r], nums[i] = nums[i], nums[r]
                r -= 1
            elif nums[i] == 0:
                nums[l], nums[i] = nums[i], nums[l]
                l += 1
                i += 1
            else:
                i += 1
